doc: add logging over RPC documentation (#22590)
nrfconnect/sdk-nrf: doc/nrf/libraries/logging/log_rpc.rst

.. _log_rpc:

Logging Remote Procedure Call
#############################

.. contents::
   :local:
   :depth: 2

> The Logging Remote Procedure Call (RPC) library provides the functionality to receive and manage log messages generated by a remote device or processor.

Reviewer: ```suggestion
The Logging Remote Procedure Call (RPC) library enables receiving and managing log messages from a remote device or processor.
```

Overview
********

> The library supports the following two main features:
> 
>   * Log streaming: receiving log messages over RPC in real-time as they are generated by the remote device.
>   * Log history: saving log messages in a ring buffer in RAM or persistent memory, and retrieving them over RPC when needed.

Reviewer: ```suggestion
The library supports the following main features:

  * Log streaming, which allows for receiving log messages over RPC in real-time as they are generated by the remote device.
  * Log history, which allows for saving log messages in a ring buffer in RAM or persistent memory, and retrieving them over RPC when needed.
```

The library also supports retrieving a core dump stored in the remote device's memory, and setting the current time used for log timestamping.

Implementation
==============

The logging RPC consists of the following components:

> * Zephyr :ref:`zephyr:logging_api` backend that implements sending log messages using :ref:`nrf_rpc`.
>   It also accepts RPC commands for managing the logging backend.
> * Log forwarder that implements receiving log messages from a remote device and passing them to the local logging subsystem.

Reviewer: ```suggestion
* Zephyr's :ref:`zephyr:logging_api` backend that implements sending log messages using :ref:`nrf_rpc`.
  It also accepts RPC commands for managing the logging backend.
* A log forwarder, which implements receiving log messages from a remote device and passing them to the local logging subsystem.
```
